Assets/MixedRealityToolkit-SDK/Features/Teleportation/MixedRealityTeleportManager.cs

```diff
@@ -0,0 +1,263 @@
+            if (CameraCache.Main.transform.parent == null)
+            {
+                var cameraParent = new GameObject("Body");
+                CameraCache.Main.transform.SetParent(cameraParent.transform);
+            }
+
+            // Make sure the camera is at the scene origin.
+            CameraCache.Main.transform.parent.transform.position = Vector3.zero;
+            CameraCache.Main.transform.localPosition = Vector3.zero;
+
+#if UNITY_EDITOR
+            if (!UnityEditor.EditorApplication.isPlaying)
+            {
+
+                var eventSystems = Object.FindObjectsOfType<EventSystem>();
+
+                if (eventSystems.Length == 0)
+                {
+                    if (!MixedRealityManager.Instance.ActiveProfile.IsInputSystemEnabled)
+                    {
+                        new GameObject("Event System").AddComponent<EventSystem>();
+                    }
+                    else
+                    {
+                        Debug.Log("The Input System didn't properly add an event system to your scene. Please make sure the Input System's priority is set higher than the teleport system.");
+                    }
+                }
+                else if (eventSystems.Length > 1)
+                {
+                    Debug.Log("Too many event systems in the scene. The Teleport System requires only one.");
+                }
+            }
+#endif // UNITY_EDITOR
+
+            TeleportDuration = MixedRealityManager.Instance.ActiveProfile.TeleportDuration;
+            teleportEventData = new TeleportEventData(EventSystem.current);
+        }
+
+        #endregion IMixedRealityManager Implementation
+
+        #region IEventSystemManager Implementation
+
+        /// <inheritdoc />
+        public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<T> eventHandler)
+        {
+            Debug.Assert(eventData != null);
+            var teleportData = ExecuteEvents.ValidateEventData<TeleportEventData>(eventData);
+            Debug.Assert(teleportData != null);
+            Debug.Assert(!teleportData.used);
+
+            // Process all the event listeners
+            base.HandleEvent(teleportData, eventHandler);
+        }
+
+        /// <summary>
+        /// Unregister a <see cref="GameObject"/> from listening to Teleport events.
+        /// </summary>
+        /// <param name="listener"></param>
+        public override void Register(GameObject listener)
+        {
+            base.Register(listener);
+        }
+
+        /// <summary>
+        /// Unregister a <see cref="GameObject"/> from listening to Teleport events.
+        /// </summary>
+        /// <param name="listener"></param>
+        public override void Unregister(GameObject listener)
+        {
+            base.Unregister(listener);
+        }
+
+        #endregion IEventSystemManager Implementation
+
+        #region IMixedRealityTeleportSystem Implementation
+
+        public float TeleportDuration
```

Comment:
summary tag. should define duration units

Reply:
I actually need to remove this. It's currently not being used.

Reply:
Scratch that. It might be used in the future. It's defined as part of the interface, and I'd rather keep it for now.